pkgs/by-name/nm/nmgui/package.nix

```diff
@@ -0,0 +1,72 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  wrapGAppsHook4,
+  gobject-introspection,
+  gtk4,
+  glib,
+  makeDesktopItem,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "nmgui";
+  version = "1.0.0";
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner = "s-adi-dev";
+    repo = "nmgui";
+    tag = "v${version}";
+    hash = "sha256-HS/n40Ng8S5N14DtEH/upwlxdzwCoOEJA40EMdCcLw4=io";
+  };
+
+  nativeBuildInputs = [
+    wrapGAppsHook4
+    gobject-introspection
+  ];
+
+  buildInputs = [
+    gtk4
+    glib
+  ];
+
+  desktopItem = makeDesktopItem {
+    name = "nmgui";
+    exec = "nmgui";
+    icon = "network-wireless-symbolic";
+    comment = "GTK4-based Network Manager GUI using nmcli";
+    desktopName = "NM GUI";
+    categories = [
+      "Network"
+      "GTK"
+    ];
+    startupNotify = true;
+  };
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,share/applications,opt/nmgui}
+    # Copy the app files
+    cp -r app $out/opt/nmgui/
+    cp -r ${desktopItem}/share/applications/* $out/share/applications/
```

Comment:
you should do `copyDesktopItems` and `desktopItems = []` instead, as dram suggested